frontend/src/components/teacher/session-creation/steps/ChooseClass.tsx

```diff
@@ -1,16 +1,146 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
-import React from "react";
-import { Text } from "@chakra-ui/react";
+import React, { useContext, useMemo } from "react";
+import { useHistory } from "react-router-dom";
+import { useQuery } from "@apollo/client";
+import { Box, Center, Flex, Text, VStack } from "@chakra-ui/react";
+
+import { GET_CLASSES_BY_TEACHER } from "../../../../APIClients/queries/ClassQueries";
+import type { ClassResponse } from "../../../../APIClients/types/ClassClientTypes";
+import { CLASSROOMS_PAGE } from "../../../../constants/Routes";
+import AuthContext from "../../../../contexts/AuthContext";
+import { getSessionStatus } from "../../../../utils/TestSessionUtils";
+import ErrorState from "../../../common/info/ErrorState";
+import LoadingState from "../../../common/info/LoadingState";
+import EmptyDistributeClassroomsMessage from "../../../common/info/messages/EmptyDistributeClassroomsMessage";
+import Pagination from "../../../common/table/Pagination";
+import usePaginatedData from "../../../common/table/usePaginatedData";
+import ClassroomCard from "../../student-management/classrooms/ClassroomCard";
 
 interface ChooseClassProps {
+  selectedClassId: string;
   setClassId: React.Dispatch<React.SetStateAction<string>>;
 }
 
-const ChooseClass = ({ setClassId }: ChooseClassProps): React.ReactElement => {
+const ChooseClass = ({
+  selectedClassId,
+  setClassId,
+}: ChooseClassProps): React.ReactElement => {
+  const { authenticatedUser } = useContext(AuthContext);
+  const { id: teacherId } = authenticatedUser ?? {};
+
+  const { loading, error, data } = useQuery<{
+    classesByTeacher: ClassResponse[];
+  }>(GET_CLASSES_BY_TEACHER, {
+    fetchPolicy: "cache-and-network",
+    variables: { teacherId },
+    skip: !teacherId,
+  });
+
+  const history = useHistory();
+  const navigateToClassrooms = () => history.push(CLASSROOMS_PAGE);
+
+  const classCards = useMemo(() => {
+    const now = new Date();
+    return data?.classesByTeacher.map(
+      ({ testSessions, students, ...classCard }) => {
+        let activeAssessments = 0;
+        testSessions.forEach((session) => {
+          if (
+            getSessionStatus(session.startDate, session.endDate, now) ===
+            "active"
+          ) {
+            activeAssessments += 1;
+          }
+        });
+
+        return {
+          ...classCard,
+          activeAssessments,
+          assessmentCount: testSessions.length,
+          studentCount: students.length,
+        };
+      },
+    );
+  }, [data]);
+
+  const { paginatedData, totalPages, currentPage, setCurrentPage } =
+    usePaginatedData(classCards);
+
   return (
-    <Text color="blue.300" textAlign="left" textStyle="header4">
-      Choose a Classroom
-    </Text>
+    <VStack align="left" spacing="2">
```

Comment:
do you think it's worth creating a wrapper for distribute flow? with the header, text and loading / error states

Reply:
donee